### DISCUSSION TITLE
typescript-react-query query variables generated type should always be optional? maybe?

### DISCUSSION THREAD

**Discussion by akafaneh** [dotansimha/graphql-code-generator#6124] [Q&A]
_Maybe i'm totally mistaken about this or i'm missing something here._

i have a query:
```typescript
const getAuthUsers = gql`
  query getAuthUsers($userId: uuid!) {
    authentication_users(where: { uuid: { _eq: $userId } }) {
      uuid
    }
  }
`;
```

and i'm trying to call the query passing the userId from a state, which can be `undefined` (have type of `string | undefined`)
```typescript
  const [authInfo] = useRecoilState(authAtoms.authInfo);
  const { data } = useGetAuthUsersQuery({ userId: authInfo?.uuid });
  //Here i get an error: Type 'string | undefined' is not assignable to type 'string'.
```

So my thoughts here for this error is that i should be able to make the whole variables object optional so i can pass the variables after checking that they are actually available
```typescript
  const variables = !!authInfo?.uuid ? { userId: authInfo?.uuid } : undefined;
  const { data } = useGetAuthUsersQuery(variables);
  //here i still get an error: Argument of type '{ userId: string; } | undefined' is not assignable to parameter of type 'Exact<{ userId: string; }>'.
```

I found the only way to get over this is make the `$userId` in my query an optional variable (remove the `!`) which feels kinda "hacky" to me, I still want the variables to be typed and passed with the right restrictions toward what is optional and what is required, but at the same time i wanna be able (NOT-to-pass)/(condition) the variables object as a whole, in case i'm waiting for some other data/state before calling the query.

My thoughts is that maybe not passing the variable object at all can prevent the query from being executed/called until the variables object is passed (with all the required properties as they will be typed as specified in the gql query operation).

i checked `@graphql-codegen/typescript-react-query` for an option maybe to enable that, and couldn't find any. now i'm thinking maybe it can an optional thing to enable/disable (the variables object being always optional)?

Or maybe i'm missing another way to do this other of what i mentioned. any help/thoughts are appreciated

Thanks.

**Comment by dotansimha** [comment]
Similar to: https://github.com/dotansimha/graphql-code-generator/issues/6014
